test/jest/config.js

```diff
@@ -0,0 +1,26 @@
+const path = require('path');
+
+module.exports = {
+  collectCoverageFrom: [
+    'ui_framework/components/**/*.js',
+    // Seems to be a bug with jest or micromatch, in which the above glob
+    // doesn't match subsequent levels of directories, making this glob necessary.
+    'ui_framework/components/**/**/*.js',
+    '!ui_framework/components/index.js',
+    '!ui_framework/components/**/*/index.js',
+  ],
+  coverageDirectory: '<rootDir>/target/test_coverage',
```

Comment:
We desperately need to sort out directory structure.  Traditionally, `build` and `target` have been reserved exclusively for the process of creating/uploading builds, so I'm not sure what breaks if we change that assumption.